packages/playwright/src/reporters/base.ts

```diff
@@ -396,6 +376,26 @@ export function formatFailure(screen: Screen, config: FullConfig, test: TestCase
   return lines.join('\n');
 }
 
+export function formatTestWarning(screen: Screen, config: FullConfig, test: TestCase, warnings: TestAnnotation[]): string[] {
+  warnings.sort((a, b) => {
+    if (!a || !a.location)
+      return 1;
+    if (!b || !b.location)
+      return -1;
+    if (a.location.line !== b.location.line)
```

Comment:
Forgot to compare `location.file`? Perhaps use "locationKey" like in another place for easier comparison?